Tribler/Core/Session.py

```diff
@@ -472,21 +472,25 @@ def start(self):
 
         return startup_deferred
 
-    def shutdown(self, checkpoint=True, gracetime=2.0, hacksessconfcheckpoint=True):
-        """ Checkpoints the session and closes it, stopping the download engine.
-        @param checkpoint Whether to checkpoint the Session state on shutdown.
-        @param gracetime Time to allow for graceful shutdown + signoff (seconds).
+    def shutdown(self):
+        """
+        Checkpoints the session and closes it, stopping the download engine.
         """
-        # Called by any thread
         deferred = self.lm.early_shutdown()
+
         def on_early_shutdown_complete(_):
             """
-            Callback that gets called when the early shutdown has been compelted.
+            Callback that gets called when the early shutdown has been completed.
             Continues the shutdown procedure that is dependant on the early shutdown.
             :param _: ignored parameter of the Deferred
             """
-            self.checkpoint_shutdown(stop=True, checkpoint=checkpoint,
-                                 gracetime=gracetime, hacksessconfcheckpoint=hacksessconfcheckpoint)
+            try:
+                self.save_session_config()
+            except Exception as e:
```

Comment:
Yeah I already saw this one in the pylint errors, will take a look. Currently, there's another bug that I have to fix first (see the tests on linux).